src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs

```diff
@@ -431,11 +431,14 @@ private int GetStatementIndex(ChildSyntaxList children, SyntaxNode statement)
             private bool DetermineIsInConstructor(SemanticDocument semanticDocument)
             {
                 if (!ContainingType.OriginalDefinition.Equals(TypeToGenerateIn.OriginalDefinition))
-                {
                     return false;
-                }
 
-                var syntaxFacts = semanticDocument.Document.GetLanguageService<ISyntaxFactsService>();
+                // If we're in an lambda/local function we're not actually 'in' the constructor.
+                // i.e. we can't actually write to read-only fields here.
+                var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+                if (SimpleNameOpt.AncestorsAndSelf().Any(n => syntaxFacts.IsAnonymousOrLocalFunction(n)))
```

Comment:
Does this need a null check? AncestorsAndSelf doesn't appear to be a extension method, whereas IsInConstructor looks like it accepts a null so it could have been fine before but isn't now.

(This question broke you into the jail of `#nullable enabling` this file.)

Reply:
it does not.  but this is one of those really null-scary files.  it basically has flow control from higher callers to ensure things.

Reply:
let me see if i can refactor this to make the null-state definitive.

Reply:
@jasonmalinowski  resolved.